src/configuration.c

```diff
@@ -181,8 +189,6 @@ void free_configuration(void)
                 free(config.strValues[i]);
```

Comment:
Hmm, there already is a call to `nc_hashmap_free` at the beginning of `read_config_from_file`. I did not see that. So we would free the hashmap whenever the configuration is reloaded. That seems reasonable.